UN-OCHA/unocha-site
# [UNO-627] Response maps

Refs: UNO-627

This PR adds a `unocha_maps` module and a `Response Map` paragraph type to be able to display a list of response under a region as a map of responses.

The map is similar to the https://reliefweb.int/disasters map and uses mapbox. 

Note: This currently uses the same tiles are RW because they seem to be the only ones available as vector tiles which is mapbox's recommendation. We may need to check if the "official" unocha mapbox tiles can be made available as vector tiles.

### Worklfow

Simply add a "Response map" paragraph to a region page  (or the homepage to have a map of all the "current responses").

A paragraph preprocess hook in the `unocha_maps` module then checks the parent node of the paragraph and if it's a region, it finds all the responses under the region in the menu hierarchy and generates a list of response nodes rendered with the `map` view mode and attach the map libraries.

This list of responses is then converted to a map via javascript on medium to large screens (it's hidden on small screens).

When javascript is not enabled for example, the list appears as a list responses.

### Styling

The styling is minimum now and done via a `uno-response-map` component.

Note:  this component being in the subtheme, the `response map` paragraps in the edit form show a list of responses instead of the map.

### Deployment

OPS-9322 will setup the token and key for the dev site. See UNO-627 for details for the local site.

### Tests

1. Checkout the branch, adds the credentials as per UNO-627
2. Recreate the image, recreate the container 
3. Import the config and clear the cache
4. Go to a region page (which has responses as children in the menu hierarchy) and add a "Response map" paragraph
5. Check that it's rendered as a map when viewing the region
